reverse_string.js

```diff
@@ -0,0 +1,12 @@
+function reverseString(s) {
+  let res = "";
+
+  for (let i = s.length - 1; i >= 0; i--) {
+    res = res + s[i]
```

Comment:
This is a working solution, but the concatenation operator (`+`) allocates a new string for every character in `s` which is actually quite expensive. JS only has immutable strings, so you can't get around that just with that data-type, but do you think you could use another datatype to do this more efficiently?